src/http-api.ts

```diff
@@ -128,6 +128,29 @@ export class HttpApi {
       }
     });
 
+    this.#api.get('/:did/query', async (req, res) => {
+      const options = {} as any;
+      for (const param in req.query as object) {
+        const keys = param.split('.');
+        const lastKey = keys.pop();
+        keys.reduce((obj, key) => obj[key] = obj[key] || {}, options)[lastKey] = req.query[param]
+      }
+      const record = await RecordsQuery.create({
+        filter: options.filter,
+        pagination: options.pagination,
+        dateSort: options.dateSort,
+      });
+      const reply = (await this.dwn.processMessage(req.params.did, record.toJSON())) as RecordsQueryReply;
+      if (reply.status.code === 200) { 
+        res.setHeader('content-type', 'application/json');
+        return res.json(reply)
+      } else if (reply.status.code === 401) {
+        return res.sendStatus(404);
+      } else {
+        return res.status(reply.status.code).send(reply);
+      }
```

Comment:
These "unhappy" paths are a bit vague to me without tests to prove the intended behavior is as coded.

Reply:
Basically this code, like the record-atomic GET we already have, either succeeds, masks the presence/non-presence of unauthorized actions as a 404 if a failure is authz-related, or defers to the error code of the DWN for all other errors.

Reply:
Those error does not make sense because we are not authenticating so there will not be 401. I updated the code.